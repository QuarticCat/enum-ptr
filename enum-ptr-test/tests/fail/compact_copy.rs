#![allow(dead_code)]

use enum_ptr::EnumPtr;

#[derive(EnumPtr)]
#[enum_ptr(copy)]
#[repr(C, usize)]
enum Foo<'a, 'b> {
    A(&'a i32),
    B(&'b i32),
}

fn main() {}
