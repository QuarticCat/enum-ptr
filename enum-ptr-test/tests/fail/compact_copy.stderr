error[E0277]: the trait bound `Foo<'a, 'b>: std::marker::Copy` is not satisfied
 --> tests/fail/compact_copy.rs:5:10
  |
5 | #[derive(EnumPtr)]
  |          ^^^^^^^ the trait `std::marker::Copy` is not implemented for `Foo<'a, 'b>`
  |
note: required by a bound in `CompactCopy`
 --> $WORKSPACE/enum-ptr/src/compact_copy.rs
  |
  |     T: From<CompactCopy<T>> + Copy,
  |                               ^^^^ required by this bound in `CompactCopy`
  = note: this error originates in the derive macro `EnumPtr` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Foo<'a, 'b>` with `#[derive(Copy)]`
  |
8 | #[derive(Copy)]
  |
